### PR TITLE
virttest: changes to support testing non-root virtiofs

### PR DESCRIPTION
**Pull request by smitterl**
utils_ids: add new module to retrieve Linux user ids

The module contains functions to retrieve uid, gid, subuid, subgid
info for a given username.

subuid, subgid are used by libvirt to run virtiofs with idmap
enabling unprivileged user access to filesystem passthrough.

Add a function to check if a given fileystem xml contains the
idmap element as expected based on user info that's retrieved
by the aforementioned functions in this module.

libvirt_xml/base:
libvirt_xml/vm_xml:
libvirt_xml/devices/filesystem_device:

Add <idmap> to the filesystem device.

<idmap> can be under /domain directly but also under //filesystem
therefore it doesn't fit below devices.

Create another folder for other elements that don't fit into
* libvirt_xml: on this level we have the different XML types like
               domain xml (vm_xml), network_xml and so on
* libvirt_xml/devices: here we have all elements that are below the
               /domain/devices node

Add properties 'gids' and 'uids' because there will be more than one.
The old properties 'gid' and 'uid' are left untouched.

Add sandbox mode to the binary subtag because it's used when running
VMs with virtiofs as non-root users.

Update some property settings functions on vm_xml to use the passed
virsh_instance when sync'ing the xml, otherwise the XML won't be sync'd
when run with an unprivileged virsh instance.

Update some property settings functions on vml_xml to use the passed
through vmxml and not sync. This will allow to chain several XML updates
and only call sync at the end externally and this way reduce the number
of virsh calls and I/O interactions improving the test performance.

Also, remove sysinfo_xml: It is more extensively defined already in
               vm_xml where it belongs due to its local use only
               below /domain. Move the get_all_processors there.

### CODE REVIEW

File: virttest/libvirt_xml/base.py
Comment (chloerh) [inline virttest/libvirt_xml/base.py:442]: I'd like to suggest that we move this class to a new file, which could similar to [character.py ](https://github.com/avocado-framework/avocado-vt/blob/c1c726a17b49c9890a44ee32ca00b96563483baa/virttest/libvirt_xml/devices/character.py)

Comment (smitterl) [reply virttest/libvirt_xml/base.py:442]: @chloerh AFAIK, "Character" is an actual base class for a device such as serial. But here the 'idmap' is not a device but a device subelement that is used by virtiofs and domain xml. Are you sure we should move this to a specific file idmap.py? I just want to double check because it seems incoherent for me.

Comment (smitterl) [reply virttest/libvirt_xml/base.py:442]: Actually the address element, common to all devices, is in devices.base. Given that idmap seems to be available on domain level, I think libvirt_xml.base is a good place for it.

Comment (chloerh) [reply virttest/libvirt_xml/base.py:442]: Oh, I understand your concern of putting it to devices/ because 'idmap' is not a device. But I also have this intuition that libvirt_xml.base used to act as a module to handle universal behaviors of all libvirt_xml subclasses, which seems not quite fit for one xml class. Maybe we could put it next to libvirt_xml/network_xml?

Comment (smitterl) [reply virttest/libvirt_xml/base.py:442]: I can do that if you insist but I have my doubts: all xml classes on the libvirt_xml level refer to a libvirt XML class, network xml is not a device but it's its own XML, not related to domain or subelements. The same holds for the other files: backup_xml, checkpoint_xml, snapshot_xml, vol_xml, aso.
The only exception I've found: sysinfo which is a subelement of vm_xml and I would actually expect it to live in that file.
I think your intuition about base.py scope I think it's right. So maybe what's missing is a subfolder 'misc' where both sysinfo_xml and idmap_xml should live?  Or we could create a file domaincommon_xml where we put both? (I got the idea from domaincommon.rng, where idmap is defined both for the domain as well as for the filesystem.)

Comment (chloerh) [reply virttest/libvirt_xml/base.py:442]: I like the idea of creating a subfolder for sysinfo_xml and idmap_xml, which could clarify the structure of all libvirt-XML classes. And I'm also ok with the domaincommon_xml idea, just a little worry that the file might become large in the future (which seem unlikely).  It's your call. Thank you for coming up with these great solutions!

Comment (smitterl) [reply virttest/libvirt_xml/base.py]: Okay, I created the 'other' subfolder. Btw I found that sysinfo was indeed already defined so I removed the sysinfo_xml file and point the reference to that instance. I'll run tests covering the idmap and sysinfo first before I re-request your review though.

Comment (smitterl) [reply virttest/libvirt_xml/base.py]: @chloerh I updated the code as required. I rerun some tests for this last change, including the sysinfo change: https://github.com/autotest/tp-libvirt/pull/5590#issuecomment-2109694624

The original class `SysinfoXML` was only used in tp-libvirt's virsh_sysinfo.

File: virttest/utils_ids.py
Comment (chunfuwen) [inline virttest/utils_ids.py:92]: this is specific to libvirt, so maybe put this file into avocado-vt/virttest/utils_libvirt folder

File: virttest/libvirt_xml/other/idmap_xml.py
Comment (chunfuwen) [inline virttest/libvirt_xml/other/idmap_xml.py:10]: I think since python3 , we recommend use absolute  path import

File: virttest/utils_ids.py
Comment (smitterl) [reply virttest/utils_ids.py]: OK, moved to new file libvirt_filesystem.py

File: virttest/libvirt_xml/other/idmap_xml.py
Comment (smitterl) [reply virttest/libvirt_xml/other/idmap_xml.py]: OK, updated

**Comment by smitterl** [comment]
@chloerh @chunfuwen I addressed your concerns, please review again. Thanks.

File: virttest/libvirt_xml/other/idmap_xml.py
Comment (chloerh) [inline virttest/libvirt_xml/other/idmap_xml.py:42]: Seems `uid` `gid` could be replaced by newly added `uids` and `gids`. If you're not going to use `uid` `gid`, I'll remove these and update the test in tp-libvirt(Seems I wrote the only case uses these 2 attrs). The current change is ok to me.

Comment (smitterl) [reply virttest/libvirt_xml/other/idmap_xml.py:42]: @chloerh Those two attributes already existed. I added the list attributes. I don't want to remove the existing attributes assuming that if they already existed, I don't want to touch them for backward compatibility.

File: virttest/libvirt_xml/__init__.py
Comment (smitterl) [reply virttest/libvirt_xml/__init__.py:43]: @chloerh thanks for finding this. seems this condition wasn't hit in my test execution. i updated the import now (VMSysinfoXML is available in libvirt_xml thanks to l.43), ref. https://github.com/autotest/tp-libvirt/pull/5590/files#diff-7f20677f24367cfbdfc37e3d39e21b61369187114c07c3a44f1ef2629e8384e2R63

File: virttest/libvirt_xml/other/idmap_xml.py
Comment (chloerh) [reply virttest/libvirt_xml/other/idmap_xml.py:42]: Actually I just want to make sure that the 2 old attributes are not used by the newly added tests. It should not block this pr, I will open another one to remove them and update existing tests. (Since I found the old 2 attrs were added by me and seems not used by other tests). Their functions overlap, that's the reason I'd like to remove the old ones. But it doesn't affect the current pr.

Comment (smitterl) [reply virttest/libvirt_xml/other/idmap_xml.py:42]: @chloerh cool, can you approve the pr?

**Review by chloerh — APPROVED** [review]

File: virttest/libvirt_xml/other/idmap_xml.py
Comment (chloerh) [reply virttest/libvirt_xml/other/idmap_xml.py:42]: Approved.

**Review by chunfuwen — APPROVED** [review]
lgtm

**Comment by yanan-fu** [comment]
@smitterl @chunfuwen  This patch introduce a new folder `other` and use the modules `idmap_xml`, but miss the `__init__.py`.    It will break all tests by default.
Could you please have a look ?  Thanks!

**Comment by smitterl** [comment]
> @smitterl @chunfuwen This patch introduce a new folder `other` and use the modules `idmap_xml`, but miss the `__init__.py`. It will break all tests by default. Could you please have a look ? Thanks!

Which tests are broken? I ran our tests and no problem.

**Comment by yanan-fu** [comment]
@smitterl  Here is the full error info:
```
avocado run --journal --vt-config /root/avocado/data/avocado-vt/backends/qemu/cfg/tests-example.cfg --force-job-id af0417dd8b69fc3eb2037c7c8bea5c58ae4a40d6 --store-logging-stream aexpect --disable-tap-job-result --vt-short-names-when-config --vt-qemu-bin /usr/libexec/qemu-kvm --vt-qemu-dst-bin /usr/libexec/qemu-kvm'
Failed to load plugin from module "avocado_vt.plugins.vt_list_archs": ModuleNotFoundError("No module named 'virttest.libvirt_xml.other'",) :
  File "/usr/local/lib/python3.6/site-packages/avocado_framework-100.1-py3.6.egg/avocado/core/extension_manager.py", line 94, in __init__
    plugin = ep.load()
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2318, in load
    return self.resolve()
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2324, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/plugins/vt_list_archs.py", line 6, in <module>
    from ..loader import arch_listing
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/loader.py", line 29, in <module>
    from .test import VirtTest
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/test.py", line 27, in <module>
    from virttest import (
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/env_process.py", line 24, in <module>
    from virttest import (
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/migration.py", line 26, in <module>
    from virttest.utils_test import libvirt
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/utils_test/libvirt.py", line 54, in <module>
    from virttest.libvirt_xml import (
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/libvirt_xml/vm_xml.py", line 14, in <module>
    from ..libvirt_xml.other.idmap_xml import VMIDMapXML

Failed to load plugin from module "avocado_vt.plugins.vt_list_guests": ModuleNotFoundError("No module named 'virttest.libvirt_xml.other'",) :
  File "/usr/local/lib/python3.6/site-packages/avocado_framework-100.1-py3.6.egg/avocado/core/extension_manager.py", line 94, in __init__
    plugin = ep.load()
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2318, in load
    return self.resolve()
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2324, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/plugins/vt_list_guests.py", line 7, in <module>
    from ..loader import guest_listing
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/loader.py", line 29, in <module>
    from .test import VirtTest
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/test.py", line 27, in <module>
    from virttest import (
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/env_process.py", line 24, in <module>
    from virttest import (
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/migration.py", line 26, in <module>
    from virttest.utils_test import libvirt
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/utils_test/libvirt.py", line 54, in <module>
    from virttest.libvirt_xml import (
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/libvirt_xml/vm_xml.py", line 14, in <module>
    from ..libvirt_xml.other.idmap_xml import VMIDMapXML

Failed to load plugin from module "avocado_vt.plugins.vt_list_archs": ModuleNotFoundError("No module named 'virttest.libvirt_xml.other'",) :
  File "/usr/local/lib/python3.6/site-packages/avocado_framework-100.1-py3.6.egg/avocado/core/extension_manager.py", line 94, in __init__
    plugin = ep.load()
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2318, in load
    return self.resolve()
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2324, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/plugins/vt_list_archs.py", line 6, in <module>
    from ..loader import arch_listing
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/loader.py", line 29, in <module>
    from .test import VirtTest
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/test.py", line 27, in <module>
    from virttest import (
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/env_process.py", line 24, in <module>
    from virttest import (
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/migration.py", line 26, in <module>
    from virttest.utils_test import libvirt
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/utils_test/libvirt.py", line 54, in <module>
    from virttest.libvirt_xml import (
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/libvirt_xml/vm_xml.py", line 14, in <module>
    from ..libvirt_xml.other.idmap_xml import VMIDMapXML

Failed to load plugin from module "avocado_vt.plugins.vt_list_guests": ModuleNotFoundError("No module named 'virttest.libvirt_xml.other'",) :
  File "/usr/local/lib/python3.6/site-packages/avocado_framework-100.1-py3.6.egg/avocado/core/extension_manager.py", line 94, in __init__
    plugin = ep.load()
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2318, in load
    return self.resolve()
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2324, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/plugins/vt_list_guests.py", line 7, in <module>
    from ..loader import guest_listing
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/loader.py", line 29, in <module>
    from .test import VirtTest
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/test.py", line 27, in <module>
    from virttest import (
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/env_process.py", line 24, in <module>
    from virttest import (
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/migration.py", line 26, in <module>
    from virttest.utils_test import libvirt
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/utils_test/libvirt.py", line 54, in <module>
    from virttest.libvirt_xml import (
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/libvirt_xml/vm_xml.py", line 14, in <module>
    from ..libvirt_xml.other.idmap_xml import VMIDMapXML

Failed to load plugin from module "avocado_vt.plugins.vt_joblock": ModuleNotFoundError("No module named 'virttest.libvirt_xml.other'",) :
  File "/usr/local/lib/python3.6/site-packages/avocado_framework-100.1-py3.6.egg/avocado/core/extension_manager.py", line 94, in __init__
    plugin = ep.load()
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2318, in load
    return self.resolve()
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2324, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/plugins/vt_joblock.py", line 18, in <module>
    from ..test import VirtTest
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/test.py", line 27, in <module>
    from virttest import (
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/env_process.py", line 24, in <module>
    from virttest import (
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/migration.py", line 26, in <module>
    from virttest.utils_test import libvirt
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/utils_test/libvirt.py", line 54, in <module>
    from virttest.libvirt_xml import (
  File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/libvirt_xml/vm_xml.py", line 14, in <module>
    from ..libvirt_xml.other.idmap_xml import VMIDMapXML

JOB ID     : af0417dd8b69fc3eb2037c7c8bea5c58ae4a40d6
JOB LOG    : /root/avocado/job-results/job-2024-05-31T02.16-af0417d/job.log

Avocado job failed: JobError: Could not find runners for runnable(s) of kind(s): avocado-vt
```

Feel free to contact me if you need the env tfor debugging.

**Comment by smitterl** [comment]
> @smitterl Here is the full error info:
> 
> ```
> avocado run --journal --vt-config /root/avocado/data/avocado-vt/backends/qemu/cfg/tests-example.cfg --force-job-id af0417dd8b69fc3eb2037c7c8bea5c58ae4a40d6 --store-logging-stream aexpect --disable-tap-job-result --vt-short-names-when-config --vt-qemu-bin /usr/libexec/qemu-kvm --vt-qemu-dst-bin /usr/libexec/qemu-kvm'
> Failed to load plugin from module "avocado_vt.plugins.vt_list_archs": ModuleNotFoundError("No module named 'virttest.libvirt_xml.other'",) :
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework-100.1-py3.6.egg/avocado/core/extension_manager.py", line 94, in __init__
>     plugin = ep.load()
>   File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2318, in load
>     return self.resolve()
>   File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2324, in resolve
>     module = __import__(self.module_name, fromlist=['__name__'], level=0)
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/plugins/vt_list_archs.py", line 6, in <module>
>     from ..loader import arch_listing
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/loader.py", line 29, in <module>
>     from .test import VirtTest
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/test.py", line 27, in <module>
>     from virttest import (
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/env_process.py", line 24, in <module>
>     from virttest import (
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/migration.py", line 26, in <module>
>     from virttest.utils_test import libvirt
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/utils_test/libvirt.py", line 54, in <module>
>     from virttest.libvirt_xml import (
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/libvirt_xml/vm_xml.py", line 14, in <module>
>     from ..libvirt_xml.other.idmap_xml import VMIDMapXML
> 
> Failed to load plugin from module "avocado_vt.plugins.vt_list_guests": ModuleNotFoundError("No module named 'virttest.libvirt_xml.other'",) :
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework-100.1-py3.6.egg/avocado/core/extension_manager.py", line 94, in __init__
>     plugin = ep.load()
>   File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2318, in load
>     return self.resolve()
>   File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2324, in resolve
>     module = __import__(self.module_name, fromlist=['__name__'], level=0)
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/plugins/vt_list_guests.py", line 7, in <module>
>     from ..loader import guest_listing
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/loader.py", line 29, in <module>
>     from .test import VirtTest
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/test.py", line 27, in <module>
>     from virttest import (
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/env_process.py", line 24, in <module>
>     from virttest import (
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/migration.py", line 26, in <module>
>     from virttest.utils_test import libvirt
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/utils_test/libvirt.py", line 54, in <module>
>     from virttest.libvirt_xml import (
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/libvirt_xml/vm_xml.py", line 14, in <module>
>     from ..libvirt_xml.other.idmap_xml import VMIDMapXML
> 
> Failed to load plugin from module "avocado_vt.plugins.vt_list_archs": ModuleNotFoundError("No module named 'virttest.libvirt_xml.other'",) :
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework-100.1-py3.6.egg/avocado/core/extension_manager.py", line 94, in __init__
>     plugin = ep.load()
>   File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2318, in load
>     return self.resolve()
>   File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2324, in resolve
>     module = __import__(self.module_name, fromlist=['__name__'], level=0)
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/plugins/vt_list_archs.py", line 6, in <module>
>     from ..loader import arch_listing
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/loader.py", line 29, in <module>
>     from .test import VirtTest
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/test.py", line 27, in <module>
>     from virttest import (
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/env_process.py", line 24, in <module>
>     from virttest import (
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/migration.py", line 26, in <module>
>     from virttest.utils_test import libvirt
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/utils_test/libvirt.py", line 54, in <module>
>     from virttest.libvirt_xml import (
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/libvirt_xml/vm_xml.py", line 14, in <module>
>     from ..libvirt_xml.other.idmap_xml import VMIDMapXML
> 
> Failed to load plugin from module "avocado_vt.plugins.vt_list_guests": ModuleNotFoundError("No module named 'virttest.libvirt_xml.other'",) :
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework-100.1-py3.6.egg/avocado/core/extension_manager.py", line 94, in __init__
>     plugin = ep.load()
>   File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2318, in load
>     return self.resolve()
>   File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2324, in resolve
>     module = __import__(self.module_name, fromlist=['__name__'], level=0)
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/plugins/vt_list_guests.py", line 7, in <module>
>     from ..loader import guest_listing
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/loader.py", line 29, in <module>
>     from .test import VirtTest
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/test.py", line 27, in <module>
>     from virttest import (
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/env_process.py", line 24, in <module>
>     from virttest import (
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/migration.py", line 26, in <module>
>     from virttest.utils_test import libvirt
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/utils_test/libvirt.py", line 54, in <module>
>     from virttest.libvirt_xml import (
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/libvirt_xml/vm_xml.py", line 14, in <module>
>     from ..libvirt_xml.other.idmap_xml import VMIDMapXML
> 
> Failed to load plugin from module "avocado_vt.plugins.vt_joblock": ModuleNotFoundError("No module named 'virttest.libvirt_xml.other'",) :
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework-100.1-py3.6.egg/avocado/core/extension_manager.py", line 94, in __init__
>     plugin = ep.load()
>   File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2318, in load
>     return self.resolve()
>   File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2324, in resolve
>     module = __import__(self.module_name, fromlist=['__name__'], level=0)
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/plugins/vt_joblock.py", line 18, in <module>
>     from ..test import VirtTest
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/avocado_vt/test.py", line 27, in <module>
>     from virttest import (
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/env_process.py", line 24, in <module>
>     from virttest import (
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/migration.py", line 26, in <module>
>     from virttest.utils_test import libvirt
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/utils_test/libvirt.py", line 54, in <module>
>     from virttest.libvirt_xml import (
>   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-104.0-py3.6.egg/virttest/libvirt_xml/vm_xml.py", line 14, in <module>
>     from ..libvirt_xml.other.idmap_xml import VMIDMapXML
> 
> JOB ID     : af0417dd8b69fc3eb2037c7c8bea5c58ae4a40d6
> JOB LOG    : /root/avocado/job-results/job-2024-05-31T02.16-af0417d/job.log
> 
> Avocado job failed: JobError: Could not find runners for runnable(s) of kind(s): avocado-vt
> ```
> 
> Feel free to contact me if you need the env tfor debugging.

Thank you, I'll create an empty __init__ file. Weird that we haven't run into this issue.

**Pull request merged**
